File: java/integracion_con_base_de_datos/guardar_un_contacto_basico/src/guardar_un_contacto_basico/guardar_un_contacto_basico.java
package guardar_un_contacto_basico;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.util.Scanner;
public class guardar_un_contacto_basico {
    @SuppressWarnings("resource")
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("=== GUARDAR CONTACTO ===");
        System.out.print("Nombre: ");
        String nombre = sc.nextLine();
        System.out.print("Teléfono: ");
        String telefono = sc.nextLine();
        System.out.print("Correo: ");
        String correo = sc.nextLine();
        Connection conn = conexion.conectar();
        if (conn == null) {
            System.out.println("❌ No se pudo conectar");
            return;
        }
        String sql = "INSERT INTO contactos (nombre, telefono, correo) VALUES (?, ?, ?)";
        try {
            PreparedStatement stmt = conn.prepareStatement(sql);
            stmt.setString(1, nombre);
            stmt.setString(2, telefono);
            stmt.setString(3, correo);
            stmt.executeUpdate();
            System.out.println("✅ Contacto guardado");
        } catch (Exception e) {
            System.out.println("❌ Error: " + e.getMessage());
        }
        sc.close();
    }
}

File: visual_basic/integracion_con_base_de_datos/guardar_un_contacto_basico/guardar_un_contacto_basico.vb
Imports System
Imports MySql.Data.MySqlClient
Module guardar_un_contacto_basico
    Sub Main()
        Console.WriteLine("📒 GUARDAR CONTACTO EN BASE DE DATOS")
        Console.Write("Nombre: ")
        Dim nombre As String = Console.ReadLine()
        Console.Write("Teléfono: ")
        Dim telefono As String = Console.ReadLine()
        Console.Write("Correo: ")
        Dim correo As String = Console.ReadLine()
        Try
            Dim conexion As MySqlConnection = ObtenerConexion()
            conexion.Open()
            Dim query As String = "INSERT INTO contactos (nombre, telefono, correo) VALUES (@nombre, @telefono, @correo)"
            Dim comando As New MySqlCommand(query, conexion)
            comando.Parameters.AddWithValue("@nombre", nombre)
            comando.Parameters.AddWithValue("@telefono", telefono)
            comando.Parameters.AddWithValue("@correo", correo)
            comando.ExecuteNonQuery()
            Console.WriteLine("✅ Contacto guardado correctamente")
            conexion.Close()
        Catch ex As Exception
            Console.WriteLine("❌ Error: " & ex.Message)
        End Try
        Console.ReadKey()
    End Sub
End Module